zellij-server/src/pty.rs

```diff
@@ -659,46 +660,52 @@ impl Pty {
                                 self.bus.senders.clone(),
                                 terminal_id,
                                 run_command.clone(),
-                            );
+                            )?;
                         } else {
-                            self.close_pane(PaneId::Terminal(terminal_id));
+                            self.close_pane(PaneId::Terminal(terminal_id))?;
                         }
                     },
                     None => {
-                        self.close_pane(PaneId::Terminal(terminal_id));
+                        self.close_pane(PaneId::Terminal(terminal_id))?;
                     },
                 },
             }
         }
         Ok(())
     }
-    pub fn close_pane(&mut self, id: PaneId) {
+    pub fn close_pane(&mut self, id: PaneId) -> Result<()>{
+        let err_context = || format!("failed to close for pane {id:?}");
         match id {
             PaneId::Terminal(id) => {
                 self.task_handles.remove(&id);
                 if let Some(child_fd) = self.id_to_child_pid.remove(&id) {
                     task::block_on(async {
+                        let err_context = || format!("failed to run async task for pane {id}");
                         self.bus
                             .os_input
                             .as_mut()
-                            .unwrap()
+                            .with_context(err_context)
+                            .fatal()
                             .kill(Pid::from_raw(child_fd))
-                            .unwrap();
+                            .with_context(err_context)
+                            .fatal();
                     });
                 }
-                self.bus.os_input.as_ref().unwrap().clear_terminal_id(id);
+                self.bus.os_input.as_ref().with_context(err_context)?.clear_terminal_id(id);
             },
             PaneId::Plugin(pid) => drop(
                 self.bus
                     .senders
                     .send_to_plugin(PluginInstruction::Unload(pid)),
             ),
         }
+        Ok(())
     }
-    pub fn close_tab(&mut self, ids: Vec<PaneId>) {
-        ids.iter().for_each(|&id| {
-            self.close_pane(id);
-        });
+    pub fn close_tab(&mut self, ids: Vec<PaneId>) -> Result<()>{
+        for id in ids {
+            self.close_pane(id)?;
```

Comment:
Could you please add an error context here?

Reply:
Yeah, I'll fix that.